File: src/proxy_kingdom.py
from requests import get

class ProxyKingdom:
	def __init__(self, token: str) -> None:
		self.api = "https://api.proxykingdom.com"
		self.headers = {
			"user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.99 Safari/537.36",
		}
		self.token = token

	def get_random_proxy(
			self,
			address: str = None,
			port: int = None,
			protocol: str = None,
			access_type: str = None,
			response_time: str = None,
			is_ssl: bool = False,
			up_time: str = None,
			country: str = None,
			continent: str = None,
			time_zone: str = None,
			last_tested: str = None) -> str:
		url = f"{self.api}/proxy?token={self.token}"
		if address:
			url += f"&address={address}"
		if port:
			url += f"&port={port}"
		if protocol:
			url += f"&protocol={protocol}"
		if access_type:
			url += f"&accessType={access_type}"
		if response_time:
			url += f"&responseTime={response_time}"
		if is_ssl:
			url += f"&isSsl={is_ssl}"
		if up_time:
			url += f"&uptime={up_time}"
		if country:
			url += f"&country={country}"
		if continent:
			url += f"&continent={continent}"
		if time_zone:
			url += f"&timezone={time_zone}"
		if last_tested:
			url += f"&lastTested={last_tested}"
		return get(
			url, headers=self.headers).json()

File: src/proxy_kingdom.js
class ProxyKingdom {
	constructor(token) {
		this.api = "https://api.proxykingdom.com"
		this.token = token
		this.headers = {
			"User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.99 Safari/537.36"
		}
	}

	async getRandomProxy(address = null, port = null, protocol = null, accessType = null, responseTime = null, isSsl = null, upTime = null, country = null, continent = null, timeZone = null, lastTested = null) {
		let url = `${this.api}/proxy?token=${self.token}`
		if (address) {
			url += `&address=${address}`
		}
		if (port) {
			url += `&port=${port}`
		}
		if (protocol) {
			url += `&protocol=${protocol}`
		}
		if (accessType) {
			url += `&accessType=${accessType}`
		}
		if (responseTime) {
			url += `&responseTime=${responseTime}`
		}
		if (isSsl) {
			url += `&isSsl=${isSsl}`
		}
		if (upTime) {
			url += `&uptime=${upTime}`
		}
		if (country) {
			url += `&country=${country}`
		}
		if (continent) {
			url += `&continent=${continent}`
		}
		if (timeZone) {
			url += `&timezone=${timeZone}`
		}
		if (lastTested) {
			url += `&lastTested=${lastTested}`
		}
		const response = await fetch(
			url, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}
}

module.exports = {ProxyKingdom}
